pkg/types/ovirt/validation/machinepool.go

```diff
@@ -1,12 +1,48 @@
 package validation
 
 import (
+	"fmt"
+
 	"k8s.io/apimachinery/pkg/util/validation/field"
 
 	"github.com/openshift/installer/pkg/types/ovirt"
+	"github.com/openshift/installer/pkg/validate"
 )
 
 // ValidateMachinePool checks that the specified machine pool is valid.
 func ValidateMachinePool(p *ovirt.MachinePool, fldPath *field.Path) field.ErrorList {
-	return field.ErrorList{}
+	allErrs := field.ErrorList{}
+
+	if p.CPU != nil {
+		if p.CPU.Cores <= 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("cores"), p.CPU.Cores, "CPU cores must be positive"))
+		}
+		if p.CPU.Sockets <= 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("sockets"), p.CPU.Sockets, "CPU sockets must be positive"))
+		}
+	}
+
+	if p.VMType != "" && !p.VMType.ValidValue() {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("vmType"), p.VMType, fmt.Sprintf("VM type must be one of %s", p.VMType.Values())))
+	}
+
+	if p.InstanceTypeID != "" {
+		if p.CPU != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("instanceTypeID"), p.InstanceTypeID, "Mixing instanceTypeID and CPU is not supported"))
+		}
+		if p.MemoryMB > 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("instanceTypeID"), p.InstanceTypeID, "Mixing instanceTypeID and Memory  is not supported"))
```

Comment:
nit: error messages should start with a lowercase letter

Reply:
done